.changeset/yellow-turtles-check.md

```diff
@@ -0,0 +1,5 @@
+---
+'@makeswift/runtime': patch
+---
+
+Wraps the Page component in a `Suspense` boundary for component registration with `React.lazy`. There were occurrences in which using a registered component with `React.lazy` would cause a hydration mismatch between the server and client.
```

Comment:
The problem was less with registered components and more with the fact that `RuntimeProvider` itself was using `React.lazy` or am I misunderstanding the issue here?

We already wrap all registered components with `<Suspense>` which is why I'm asking.

Reply:
Ah, yeah I was misunderstanding it here.